deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -964,6 +966,22 @@ async def get_notification_messages(
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+@app.post("/notification_messages", response_model=ViewNotificationMessageResponse)
```

Comment:
since this is updating a row should it be a put instead of a post?

Reply:
actually, since this method isn't idempotent maybe it makes more sense as a post

Reply:
This one is a tough one to decide, and I always get them wrong anyways.   The [mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) say:

> The POST method submits an entity to the specified resource, often causing a change in state or side effects on the server.
> The PUT method replaces all current representations of the target resource with the request content.


I'm inclined to leave this as a POST.